local/compose.go

```diff
@@ -115,56 +121,127 @@ func getContainerName(c moby.Container) string {
 }
 
 func (s *composeService) Down(ctx context.Context, projectName string) error {
-	list, err := s.apiClient.ContainerList(ctx, moby.ContainerListOptions{
+	eg, _ := errgroup.WithContext(ctx)
+	w := progress.ContextWriter(ctx)
+
+	project, err := s.projectFromContainerLabels(ctx, projectName)
+	if err != nil || project == nil {
+		return err
+	}
+
+	err = InReverseDependencyOrder(ctx, project, func(c context.Context, service types.ServiceConfig) error {
+		filter := filters.NewArgs(projectFilter(project.Name), serviceFilter(service.Name))
```

Comment:
tl;dr: I'll look into this "detail" later, just a note for myself

to avoid many calls to `ContainerList` API with service filter we could list all project containers and filter the returned slice. 

I guess this would be a natural design if one could write something like
```go
containers.filter (c) -> c.Labels[serviceLabel] == service.Name
```
but thanks to Golang it is simpler to invoke API multiple times to avoid 10 extra line of boiler plate slice-filterting code